### PR TITLE
Validation updates and fixes for maxAnisotropy values

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
Two commits:
1) IMG specific VUID implementation
 VUID 12600872 (must disable anisotropy for CUBIC_IMG filters)

2) Several related new/clarified maxAniso changes:
VUID-VkSamplerCreateInfo-unnormalizedCoordinates-01076/VALIDATION_ERROR_12600868:
    Must disable anisotropy when unnormalized coordinates are enabled.
VUID-VkSamplerCreateInfo-anisotropyEnable-01071/VALIDATION_ERROR_1260085e:
    maxAnisotropy must be in range [1.0, maxSamplerAnisotropy]
VUID-VkSamplerCreateInfo-anisotropyEnable-01070/VALIDATION_ERROR_1260085c
    anisotropyEnable must be VK_FALSE if samplerAnisotropy feature is not enabled.

Removed test for: "samplerAnisotropy feature was not enabled ...
maxAnisotropy member ... must be 1.0" to conform with the resolution of
https://github.com/KhronosGroup/Vulkan-Docs/issues/503.  Fixes https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/issues/2166

### CODE REVIEW

File: layers/parameter_validation_utils.cpp [hunk 1]
```diff
@@ -1475,12 +1481,34 @@ bool pv_vkCreateSampler(VkDevice device, const VkSamplerCreateInfo *pCreateInfo,
...
-                            "vkCreateSampler(): The samplerAnisotropy feature was not enabled at device-creation time, so the "
-                            "maxAnisotropy member of the VkSamplerCreateInfo structure must be 1.0 but is %f.",
-                            pCreateInfo->maxAnisotropy);
+        const auto &features = device_data->physical_device_features;
+        const auto &limits = device_data->device_limits;
+        if (pCreateInfo->anisotropyEnable == VK_TRUE) {
+            if (!in_inclusive_range(pCreateInfo->maxAnisotropy, 1.0F, limits.maxSamplerAnisotropy)) {
+                skip |= log_msg(report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_UNKNOWN_EXT, 0, __LINE__,
+                                VALIDATION_ERROR_1260085e, LayerName,
+                                "vkCreateSampler: value of %s must be in range [1.0, %f] %s, but %f found. %s",
```
Comment (tobine) [inline layers/parameter_validation_utils.cpp]: Prefer "()" on end of function name for clarity and consistency. It's not 100% in the layers, but for the most part the function prefix is of the format "vkFuncName():". Same for lines 1500 & 1508 below.

File: layers/parameter_validation_utils.cpp [hunk 2]
```diff
@@ -1539,6 +1539,18 @@ bool pv_vkCreateSampler(VkDevice device, const VkSamplerCreateInfo *pCreateInfo,
...
                         validation_error_map[VALIDATION_ERROR_1260086e]);
         }
+
+        // Checks for the IMG cubic filtering extension
+        if (device_data->extensions.vk_img_filter_cubic) {
+            if ((pCreateInfo->anisotropyEnable == VK_TRUE) &&
+                ((pCreateInfo->minFilter == VK_FILTER_CUBIC_IMG) || (pCreateInfo->magFilter == VK_FILTER_CUBIC_IMG))) {
+                skip |= log_msg(report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_UNKNOWN_EXT, 0, __LINE__,
+                                VALIDATION_ERROR_12600872, LayerName,
+                                "vkCreateSampler: Anisotropic sampling must not be VK_TRUE when either minFilter or magFilter are "
```
Comment (tobine) [inline layers/parameter_validation_utils.cpp]: Same "()" comment as previous.

File: tests/layer_validation_tests.cpp [hunk 3]
```diff
@@ -1209,6 +1227,77 @@ TEST_F(VkLayerTest, MirrorClampToEdgeNotEnabled) {
...
+        VkSampler sampler = VK_NULL_HANDLE;
+        m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT, code);
+        err = vkCreateSampler(m_device->device(), pCreateInfo, NULL, &sampler);
+        m_errorMonitor->VerifyFound();
+        if (VK_SUCCESS == err) {
+            vkDestroySampler(m_device->device(), sampler, NULL);
+        }
+    };
+
+    // maxIsotropy out-of-bounds low.
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: maxAnisotropy? (also a few other places)

Comment (jzulauf-lunarg) [reply tests/layer_validation_tests.cpp]: Good catch, but I only see two examples of 'maxiso' strings

File: tests/layer_validation_tests.cpp [hunk 4]
```diff
@@ -1209,6 +1227,96 @@ TEST_F(VkLayerTest, MirrorClampToEdgeNotEnabled) {
...
+    sampler_info.maxAnisotropy = NearestNonEqual(1.0F, NearestLessThan());
+    do_test(VALIDATION_ERROR_1260085e, &sampler_info);
+    sampler_info.maxAnisotropy = sampler_info_ref.maxAnisotropy;
+
+    // maxIsotropy out-of-bounds high.
+    sampler_info.maxAnisotropy = NearestNonEqual(m_device->phy().properties().limits.maxSamplerAnisotropy, NearestGreaterThan());
+    do_test(VALIDATION_ERROR_1260085e, &sampler_info);
+    sampler_info.maxAnisotropy = sampler_info_ref.maxAnisotropy;
+
+    // Both isotropy and unnormalize coords enabled
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: aniso

Comment (jzulauf-lunarg) [reply tests/layer_validation_tests.cpp]: unormalized<--- with a d

**Pull request merged**